test/release/examples/primers/interopWithC.chpl

```diff
@@ -0,0 +1,332 @@
+
+   Chapel as a Library
+   -------------------
+
+   .. _primers-C-interop-functions:
+
+   Symbol Availability
+   +++++++++++++++++++
+
+*/
+
+/*
+   When creating a library file from Chapel code, only those symbols with
+   ``export`` attached to them will be available from outside the library.
+*/
+
+module interopWithC {
+
+   export proc foo() { // foo will be available to outside code
+      writeln("Called foo()");
+   }
+
+   proc bar() { // bar will not be available to outside code
+      writeln("Called bar()");
+   }
+
+
+/*
+   At present, only functions can be exported from Chapel code.
+*/
+
+/*
+   .. _primers-C-interop-compiling-library:
+
+   Compiling the Chapel Library
+   ++++++++++++++++++++++++++++
+*/
+
+/*
+   To generate a library from a Chapel code file, compile the Chapel file with
+   the ``--library`` flag.  E.g.
+*/   
+
+/*
+   .. code-block:: bash
+
+      chpl --library interopWithC.chpl
+*/
+
+/*
+   There are various flags and settings that can be used to modify the library
+   that gets created and create additional helper files.  These are listed in
+   detail in the :ref:`readme-libraries` Technical Note.  For this primer, we
+   will focus on the default settings and ``--static`` compilation - this will
+   rely on the generated library being named ``libinteropWithC.a``, with a
+   generated header file named ``interopWithC.h``, both of which live in a
+   generated ``lib`` directory.
+   Since ``--static`` is not supported on MacOS you will have to dynamically link
+   the generated library. Please refer to :ref:`readme-libraries` for instructions
+   on how to do that.
+*/
+
+/*
+   .. _primers-C-interop-using-library:
+
+   Using the Chapel Library
+   ++++++++++++++++++++++++
+*/
+
+/*
+   To access the symbols in the generated Chapel library from a C program,
+   ``#include`` the generated header file:
+*/ 
+
+/*
+   .. literalinclude:: ../../../test/release/examples/primers/cClient.test.c
+      :language: C
+      :lines: 1-2
+*/
+
+/*   
+   Before calling any of the functions defined by the library, the Chapel
+   runtime and internal libraries must be initialized.  This is done by calling
+   ``chpl_library_init``, which is defined in
+   ``$CHPL_HOME/runtime/include/chpl-init.h`` and accessible with the above
+   ``#include``.
+*/ 
+
+/*
+   .. literalinclude:: ../../../test/release/examples/primers/cClient.test.c
+      :language: C
+      :lines: 5-7
+*/
+
+/*   
+   Then, if the exported functions rely upon any global variables, the module
+   initialization function declared in the header file must be called.  If there
+   is uncertainty about whether doing so is necessary, it is recommended to call
+   this function.  In this example, this call occurs like this:
+*/ 
+
+/*
+   .. literalinclude:: ../../../test/release/examples/primers/cClient.test.c
+      :language: C
+      :lines: 8-9
+*/
+
+//
+// After that, the exported functions may be called:
```

Comment:
Maybe mention here that we'll see the definitions of the unfamiliar exported functions later?